### ISSUE TITLE
Trained weight

### ISSUE THREAD

**Issue by ShuvenduRoy** [LynnHo/AttGAN-Tensorflow#2]
I appreciate your approach to share the code for this amazing paper. Would you be kind enough to share the trained weights of the model

**Comment by LynnHo** [comment]
@ShuvenduBikash Yes, we will upload the well trained models later.

**Comment by AidasK** [comment]
Any updates?

**Comment by LynnHo** [comment]
@AidasK As soon as possible.

**Comment by AidasK** [comment]
@LynnHo How long it takes for you to complete this training? Are you running on multiple GPU's?

I tried running 128 images and it takes 4-5s per step on Nvidia Tesla K80. So it should take 947*200*4 / 60 / 60 /24 ~= 9 days

Though quality is very low and I am wondering how long it should take with HD images 512x512 (80 days?)

**Comment by LynnHo** [comment]
@AidasK I train the 128x128 network on GTX 1080 Ti, about 1-2s per step and less than two days to finish the training. The largest size I have tried is 384x384, and the training can be stopped at 4-5 days. The well-trained models will be upload in this week. You can also try a smaller network for faster training.

**Comment by LynnHo** [comment]
@ShuvenduBikash @AidasK the well-trained models are available [here](https://drive.google.com/drive/folders/11uEWeQNLAs7eOPcZkfKkraQuYS-LGz_7), the 128x128 model is uploaded, and the 384x384 models will be uploaded later.

**Comment by AidasK** [comment]
Thanks! Any success with 384x384?

**Comment by AidasK** [comment]
384x384 results are far from good. Attributes such as glasses and bold hardly works. What do you think is the cause? p.s. thanks for uploading the model and maintaining this project 👍

**Comment by LynnHo** [comment]
@AidasK bold and glasses are really hard for existing methods. I don't exactly know the true reason why these attributes are so hard, but I guess that it is caused by the need of large appearance modification to edit these attributes. Do you have any idea? Besides, the 384-hd model has been uploaded (need [HD-Celeba](https://github.com/LynnHo/HD-CelebA-Cropper)).

**Comment by AidasK** [comment]
Maybe a loss function is not suitable for these attributes. I have modified this code a bit to be able to run it on any image and these are my results:
My image:
![182638](https://user-images.githubusercontent.com/2088484/41658364-cf4c4cae-749e-11e8-9073-aadf30b9b317.png)
Random celebA hd image:
![182639](https://user-images.githubusercontent.com/2088484/41658365-cf6a1aae-749e-11e8-96bd-c8e7b07cc7b2.png)
I am searching for a way how to achieve https://www.faceapp.com/ quality, but attgan is far far away from it. Maybe you know projects/papers who has achieved such results?

**Comment by LynnHo** [comment]
@AidasK Which model did you use? Did you give a correct attribute vector for your own image (it's very important)?

I have tried the faceapp just now. I am not sure but I think it deals different attributes with different models (AttGAN uses only one), maybe CycleGAN? The quality for smile and eyeglasses is good, but for the other attributes such as age, mustache and gender, I think AttGAN is actually better. AttGAN is a research paper, but I think AttGAN will also perform well on many attributes with engineering effort.

Here are some special papers for attribute editing:
    1. Facelet-Bank - high resolution. [Paper](http://arxiv.org/abs/1803.05576) [Code](https://github.com/yingcong/Facelet_Bank)
    2. ST-GAN - good for eyeglasses. [Paper](https://arxiv.org/abs/1803.01837) [Code](https://github.com/chenhsuanlin/spatial-transformer-GAN)
They have superior advantages and quality on some aspects, but they also have much limitation. I know many other methods for attribute editing, but to my knowledge, none of them handle all attributes with pretty good quality. Facial attribute editing still needs research effort.

Papers I know:
GAN based:
    1. [Sparsely Grouped Multi-task Generative Adversarial Networks for Facial Attribute Manipulation](http://arxiv.org/abs/1805.07509)
    2. [Mask-aware Photorealistic Face Attribute Manipulation](http://arxiv.org/abs/1804.08882)
    3. [Composable Unpaired Image to Image Translation](http://arxiv.org/abs/1804.05470)
    4. [Modular Generative Adversarial Networks](http://arxiv.org/abs/1804.03343)
    5. [RSGAN: Face Swapping and Editing using Face and Hair Representation in Latent Spaces](http://arxiv.org/abs/1804.03447)
    6. [ELEGANT: Exchanging Latent Encodings with GAN for Transferring Multiple Face Attributes](http://arxiv.org/abs/1803.10562)
    7. [Image Generation and Translation with Disentangled Representations](http://arxiv.org/abs/1803.10567)
    8. [StarGAN: Unified Generative Adversarial Networks for Multi-Domain Image-to-Image Translation](http://arxiv.org/abs/1711.09020)
    9. [Learning Compositional Visual Concepts with Mutual Consistency](http://arxiv.org/abs/1711.06148)
    10. [DNA-GAN: Learning Disentangled Representations from Multi-Attribute Images](http://arxiv.org/abs/1711.05415)
    11. [Unsupervised Visual Attribute Transfer with Reconfigurable Generative Adversarial Networks](http://arxiv.org/abs/1707.09798)
    12. [Fader Networks: Manipulating Images by Sliding Attributes](http://arxiv.org/abs/1706.00409)
    13. [GeneGAN: Learning Object Transfiguration and Attribute Subspace from Unpaired Data](http://arxiv.org/abs/1705.04932)
    14. [Unsupervised Image-to-Image Translation Networks](http://arxiv.org/abs/1703.00848)
    15. [Learning Residual Images for Face Attribute Manipulation](http://arxiv.org/abs/1612.05363)
    16. [Deep Identity-aware Transfer of Facial Attributes](http://arxiv.org/abs/1610.05586)
    17. [Invertible Conditional GANs for image editing](https://arxiv.org/abs/1611.06355)
    18. [Autoencoding beyond pixels using a learned similarity metric](http://arxiv.org/abs/1512.09300)

Others:
    1. [Facelet-Bank for Fast Portrait Manipulation](http://arxiv.org/abs/1803.05576)
    2. [Deep Feature Interpolation for Image Content Changes](http://arxiv.org/abs/1611.05507)
    3. [Convolutional Network for Attribute-driven and Identity-preserving Human Face Generation](https://arxiv.org/abs/1608.06434)

**Comment by AidasK** [comment]
Thanks for this huge list, found some new stuff. I have tried Facelet-Bank before and it's as close as it gets to faceapp features. 
And yes, I did some experiments with attributes. It has a huge impact on image quality, but it's still far from good. Also tried different face alignment, but situation doesn't change much.
